### ISSUE TITLE
Noise audio files

### ISSUE THREAD

**Issue by EmreOzkose** [k2-fsa/icefall#422]
Hi,

I use pre-trained pruned_transducer_stateless3/greedy search model to infer a noise file ([noise.zip](https://github.com/k2-fsa/icefall/files/8900619/dtmf.zip)). I expect nothing, because there is not any speech, but I got ` O`. Do you have any advise to handle these situations?

**Comment by csukuangfj** [comment]
Can you post the decoding logs?

**Comment by EmreOzkose** [comment]
```
(k2l2) yunusemre.ozkose@server1:/path/to/icefall/egs/librispeech/ASR$ ./pruned_transducer_stateless3/pretrained.py --checkpoint /path/to/icefall-asr-librispeech-pruned-transducer-stateless3-2022-05-13/exp/pretrained-iter-1224000-avg-14.pt --bpe-model /path/to/icefall-asr-librispeech-pruned-transducer-stateless3-2022-05-13/data/lang_bpe_500/bpe.model --method greedy_search /path/to/audio/dtmf.wav 
2022-06-15 11:31:03,331 INFO [pretrained.py:225] {'best_train_loss': inf, 'best_valid_loss': inf, 'best_train_epoch': -1, 'best_valid_epoch': -1, 'batch_idx_train': 0, 'log_interval': 50, 'reset_interval': 200, 'valid_interval': 3000, 'feature_dim': 80, 'subsampling_factor': 4, 'encoder_dim': 512, 'nhead': 8, 'dim_feedforward': 2048, 'num_encoder_layers': 12, 'decoder_dim': 512, 'joiner_dim': 512, 'model_warm_step': 3000, 'env_info': {'k2-version': '1.15.1', 'k2-build-type': 'Release', 'k2-with-cuda': True, 'k2-git-sha1': 'b173c11ba379e2da0056281fe6b2d56f081419be', 'k2-git-date': 'Mon Apr 18 16:10:45 2022', 'lhotse-version': '1.4.0.dev+git.21b5b3f.clean', 'torch-version': '1.10.1', 'torch-cuda-available': True, 'torch-cuda-version': '11.1', 'python-version': '3.8', 'icefall-git-branch': 'master', 'icefall-git-sha1': 'dbda164-clean', 'icefall-git-date': 'Thu Jun 9 06:42:18 2022', 'icefall-path': '/path/to/icefall', 'k2-path': '/path/to/miniconda3/envs/k2l2/lib/python3.8/site-packages/k2/__init__.py', 'lhotse-path': '/path/to/miniconda3/envs/k2l2/lib/python3.8/site-packages/lhotse/__init__.py', 'hostname': 'boxx-3', 'IP address': 'server1'}, 'checkpoint': '/path/to/icefall-asr-librispeech-pruned-transducer-stateless3-2022-05-13/exp/pretrained-iter-1224000-avg-14.pt', 'bpe_model': '/path/to/icefall-asr-librispeech-pruned-transducer-stateless3-2022-05-13/data/lang_bpe_500/bpe.model', 'method': 'greedy_search', 'sound_files': ['/path/to/audio/dtmf.wav'], 'sample_rate': 16000, 'beam_size': 4, 'beam': 4, 'max_contexts': 4, 'max_states': 8, 'context_size': 2, 'max_sym_per_frame': 1, 'blank_id': 0, 'unk_id': 2, 'vocab_size': 500}
2022-06-15 11:31:03,331 INFO [pretrained.py:231] device: cuda:0
2022-06-15 11:31:03,331 INFO [pretrained.py:233] Creating model
2022-06-15 11:31:03,812 INFO [pretrained.py:237] Number of model parameters: 80199888
2022-06-15 11:31:07,236 INFO [pretrained.py:245] Constructing Fbank computer
2022-06-15 11:31:07,238 INFO [pretrained.py:255] Reading sound files: ['/path/to/audio/dtmf.wav']
2022-06-15 11:31:07,395 INFO [pretrained.py:261] Decoding started
2022-06-15 11:31:07,992 INFO [pretrained.py:280] Using greedy_search
2022-06-15 11:31:08,005 INFO [pretrained.py:339] 
/path/to/audio/dtmf.wav:
O


2022-06-15 11:31:08,005 INFO [pretrained.py:341] Decoding Done

```

**Comment by ahazned** [comment]
In my trials using beam search instead of greedy search can output "nothing" instead of speech when the input audio is just dtmf or enviromental noise, but I wonder what is the best strategy to handle these noises in models like rnn-t's? In good old hybrid models we use fillers that are both included in the lexicon&supervisions and they can handle such noises, but in icefall I think we only depend on Musan augmentations for the model to learn such patterns.

**Comment by csukuangfj** [comment]
0 is not a valid word in `data/lang_bpe_500/words.txt`. There must be something wrong. I am looking into it.

**Comment by csukuangfj** [comment]
Ok, it decodes to the big `O` with token ID 152. I just tried all the supported decoding methods and all decode to `O`.

**Comment by csukuangfj** [comment]
Here is what @danpovey  replies:

> Neural nets.can easily give weird results for out of distribution data

> Perhaps we can predict feats from embedding output, with extremely.small loss weight so it does not pass them through.  If feat diff is very large, could say it is OOD.?

> Actually,.can just detach() embedding output, no need for very small weight then.

**Comment by EmreOzkose** [comment]
Thank you
